packages/slice-machine/src/modules/pushChangesSaga/index.ts

```diff
@@ -1,281 +1,186 @@
-//     })
-//   );
-// }
-
-// function displayGeneralError() {
-//   return put(
-//     openToasterCreator({
-//       message:
-//         "An unexpected error happened while contacting the Prismic API, please try again or contact us.",
-//       type: ToasterType.ERROR,
-//     })
-//   );
-// }
-
-// function* watchChangesPush() {
-//   yield takeLatest(
-//     getType(changesPushCreator),
-//     withLoader(changesPushSaga, LoadingKeysEnum.CHANGES_PUSH)
-//   );
-// }
-
-// // Saga Exports
-// export function* watchChangesPushSagas() {
-//   yield fork(watchChangesPush);
-// }
+import { getState, pushChanges } from "../../apiClient";
+import {
+  call,
+  fork,
+  put,
+  SagaReturnType,
+  takeLatest,
+} from "redux-saga/effects";
+import {
+  ActionType,
+  createAsyncAction,
+  getType,
+  Reducer,
+} from "typesafe-actions";
+import { withLoader } from "../loading";
+import { openToasterCreator, ToasterType } from "../toaster";
+import { modalOpenCreator } from "../modal";
+import { ModalKeysEnum } from "../modal/types";
+import { LoadingKeysEnum } from "../loading/types";
+import { refreshStateCreator } from "../environment";
+import {
+  InvalidCustomTypeResponse,
+  PushChangesPayload,
+} from "@lib/models/common/TransactionalPush";
+import { ComponentUI } from "@lib/models/common/ComponentUI";
+import { CustomTypeSM } from "@lib/models/common/CustomType";
+import { trackPushChangesSuccess } from "./trackPushChangesSuccess";
+import Tracker from "@src/tracking/client";
+import { SliceMachineManagerClient } from "@slicemachine/manager/client";
+import { ChangesStatus } from "@lib/models/common/ModelStatus";
+
+export type ChangesPushSagaPayload = PushChangesPayload & {
+  changedSlices: ReadonlyArray<{ c: ComponentUI; status: ChangesStatus }>;
+  changedCustomTypes: ReadonlyArray<{
+    c: CustomTypeSM;
+    status: ChangesStatus;
+  }>;
+};
+
+type Limit = NonNullable<
+  Awaited<
+    ReturnType<SliceMachineManagerClient["transactionalMerge"]["pushChanges"]>
+  >
+>;
+
+export const changesPushCreator = createAsyncAction(
+  "PUSH_CHANGES.REQUEST",
+  "PUSH_CHANGES.RESPONSE",
+  "PUSH_CHANGES.FAILURE"
+)<ChangesPushSagaPayload, undefined, Limit | InvalidCustomTypeResponse>();
+
+export const sortDocumentLimits = (limit: Readonly<Limit>) => ({
+  ...limit,
+  details: {
+    ...limit.details,
+    customTypes: [...limit.details.customTypes].sort(
+      (doc1, doc2) => doc2.numberOfDocuments - doc1.numberOfDocuments
+    ),
+  },
+});
+
+const MODAL_KEY_MAP = {
+  // INVALID_CUSTOM_TYPES: ModalKeysEnum.REFERENCES_MISSING_DRAWER,
+  ["SOFT"]: ModalKeysEnum.SOFT_DELETE_DOCUMENTS_DRAWER,
+  ["HARD"]: ModalKeysEnum.HARD_DELETE_DOCUMENTS_DRAWER,
+};
+
+export function* changesPushSaga({
+  payload,
+}: ReturnType<typeof changesPushCreator.request>): Generator {
+  const startTime = Date.now();
+  const { changedSlices, changedCustomTypes } = payload;
+
+  const sliceChanges: Parameters<typeof pushChanges>[0]["changes"] =
```

Comment:
The types don't look great.`Parameters<typeof pushChanges>[0]["changes"]`. Could we let Typescript infer them?